### PR TITLE
Add some useful helpers to Response.cfc

### PR DESCRIPTION
**Pull request by gpickin**
Propose some helpful helpers for the Response object.
I know some of these might be too much, but I thought better to show most of them, and we can cut it out.

### CODE REVIEW

File: system/web/context/Response.cfc [hunk 1]
```diff
@@ -149,6 +215,63 @@ component accessors="true" {
...
+	 *
+	 * @return Returns the Response object for chaining
+	 */
+	function setErrorMessage( required errorMessage, statusCode, statusText ) {
+		setError( true );
+		addMessage( arguments.errorMessage );
+		if ( !isNull( arguments.statusCode ) ) {
+			if ( !isNull( arguments.statusText ) ) {
+				settatusCode( arguments.statusCode, arguments.statusText );
+			} else {
```
Comment (elpete) [inline system/web/context/Response.cfc:267]: Typo here.

File: system/web/context/Response.cfc [hunk 2]
```diff
@@ -112,6 +112,72 @@ component accessors="true" {
 	 * The headers to send with the response
 	 */
 	property name="headers" type="array";
+	
+	STATUS_TEXTS = {
```
Comment (lmajano) [inline system/web/context/Response.cfc:116]: These already exist in the request context? why duplicate them here

File: system/web/context/Response.cfc [hunk 3]
```diff
@@ -149,6 +215,63 @@ component accessors="true" {
...
 	}
+	
+	/**
+	 * Sets the status code with a statusText for the API response
+	 * @statusCode The status code to be set
+	 * @statusText The status text to be set
+	 *
+	 * @return Returns the Response object for chaining
+	 */
+	function setStatusCode( required statusCode, statusText ) {
```
Comment (lmajano) [inline system/web/context/Response.cfc:226]: Since this overrides the setter for the `statusCode` why not calling just `setStatus()`

Comment (lmajano) [reply system/web/context/Response.cfc:226]: You can then shorten the arguments to :`code` and `text`

File: system/web/context/Response.cfc [hunk 4]
```diff
@@ -149,6 +215,63 @@ component accessors="true" {
...
+	 *
+	 * @data           The struct containing both results and pagination.
+	 * @resultsKey     The name of the key with the results.
+	 * @paginationKey  The name of the key with the pagination.
+	 *
+	 * @return Returns the Response object for chaining
+	 */
+	function setDataWithPagination( data, resultsKey = "results", paginationKey = "pagination" ) {
+		variables.data = arguments.data[ arguments.resultsKey ];
+		variables.pagination = arguments.data[ arguments.paginationKey ];
```
Comment (lmajano) [inline system/web/context/Response.cfc:250]: use the `setPagination()` here, so we can get consistency in the struct being sent, also to make it ACF safe.

**Comment by lmajano** [comment]
@gpickin if you want these before our beta 2 by ITB, add your changes

**Comment by lmajano** [comment]
Merged manually

**Pull request closed without merging**